### PR TITLE
https://issues.redhat.com/browse/ACM-4191 Updating max number of managed clusters

### PR DESCRIPTION
**Pull request by dockerymick**
https://issues.redhat.com/browse/ACM-4191

### CODE REVIEW

**Review by JakobGray — COMMENTED** [review]
Is this PR just to add an empty column?

**Comment by dockerymick** [comment]
@JakobGray No the PR is for updating the table with the numerical value that represent the maximum clusters. I added SNO for nodes because it was mentioned in the issue. In the issue, it is mentioned to be the smallest type of cluster. I can remove the SNO row, but need the maximum cluster info

File: install/perform_scale.adoc [hunk 1]
```diff
@@ -43,6 +44,17 @@ The following table shows the configuration information for the clusters on the
...
+
+| single node OpenShift (SNO)
+|
+|
+|
+|
+|
+|
+|
+|
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion

```

File: install/perform_scale.adoc [hunk 2]
```diff
@@ -24,7 +24,7 @@ The maximum number of clusters that {product-title-short} can manage varies base
 The following table shows the configuration information for the clusters on the Amazon Web Services cloud platform that were used during this testing:
 
 |===
-| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region
+| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region | Tested maximum clusters
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region | Tested maximum managed clusters
```

File: install/perform_scale.adoc [hunk 3]
```diff
@@ -34,6 +34,7 @@ The following table shows the configuration information for the clusters on the
 | 100
 | 3
 | us-east-1
+|
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
| 3500
```

File: install/perform_scale.adoc [hunk 4]
```diff
@@ -43,6 +44,7 @@ The following table shows the configuration information for the clusters on the
 | 100
 | 3 or 5 nodes
 | us-east-1
+|
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
| 3500
```

**Comment by dockerymick** [comment]
@JakobGray @bweidenb I made updates to the columns

**Comment by dockerymick** [comment]
@JakobGray @bweidenb when you both have an opportunity, will you review this PR?

**Comment by dockerymick** [comment]
"PR looks good. This PM signing off." @bweidenb 
Screenshot 2023-08-29 at 11 00 39 AM

@JakobGray can you confirm this information? Bradd shared a link to a blog that mentioned 2500 pods. Should anything in the PR be changed based on the provided info?

**Comment by JakobGray** [comment]
@dockerymick I've never done anything related to scale testing before so I can't personally confirm the maximum number of managed clusters supported. I believe @akrzos has.

**Comment by dockerymick** [comment]
@akrzos when you have an opportunity, will you review the PR?

**Comment by akrzos** [comment]
I disagree with this PR, while the maximum tested value was 3500, it was not performed on AWS with those instance sizes.  Rather it was tested and still is tested to this day in the scale lab using bare metal machines to host the hub ACM cluster.

File: install/perform_scale.adoc [hunk 5]
```diff
@@ -43,6 +44,7 @@ The following table shows the configuration information for the clusters on the
 | 100
 | 3 or 5 nodes
 | us-east-1
+| 3500
 |===
```
Comment (dockerymick) [inline install/perform_scale.adoc:44]: @akrzos Thanks for sharing your feedback. Is your suggestion to change the column name "Tested maximum managed clusters"

Comment (akrzos) [reply install/perform_scale.adoc:44]: No. I am saying that the table is suggesting those instance sizes are what was tested for that maximum managed cluster count which is **not** the case.  The maximum count here was tested using bare metal machines and **not** on AWS.

File: install/perform_scale.adoc [hunk 6]
```diff
@@ -21,10 +21,10 @@ The maximum number of clusters that {product-title-short} can manage varies base
 * Number of resources in the cluster, which depends on factors like the number of policies and applications that are deployed.
 * Configuration of the hub cluster, such as how many pods are used for scaling.
 
-The following table shows the configuration information for the clusters on the Amazon Web Services cloud platform that were used during this testing:
+The following table shows the configuration information for the clusters on bare metal enviornments that were used during this testing:
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
The following table shows the configuration information for the clusters on bare metal environments that were used during this testing:
```

File: install/perform_scale.adoc (hunk 5 above)
Comment (dockerymick) [reply install/perform_scale.adoc:44]: @akrzos thanks again Alex, I clarified the statement. Let me know your thoughts

**Comment by dockerymick** [comment]
@akrzos will you re-review please?

File: install/perform_scale.adoc [hunk 7]
```diff
@@ -21,10 +21,12 @@ The maximum number of clusters that {product-title-short} can manage varies base
...
 
-The following table shows the configuration information for the clusters on the Amazon Web Services cloud platform that were used during this testing:
-
+The following table shows the configuration information for SNO clusters on bare metal environments that were used during this testing:
+//Suggesting to add hardware - Alex 
+//Need to confirm this w/Joydeep | MJ | 09/15/23
+//What info can be provided for 100 clusters or 200 clusters
 |===
-| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region
+| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region | Tested maximum SNO managed clusters
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region | Tested maximum managed clusters
```

File: install/perform_scale.adoc [hunk 8]
```diff
@@ -43,6 +46,7 @@ The following table shows the configuration information for the clusters on the
 | 100
 | 3 or 5 nodes
 | us-east-1
+| 3500
 |===
 
 For more information about infrastructure nodes, see, xref:../install/install_connected.adoc#installing-on-infra-node[Installing the {product-title-short} hub cluster on infrastructure nodes]. Also see link:https://docs.openshift.com/container-platform/4.11/machine_management/creating-infrastructure-machinesets.html[Creating infrastructure machine sets].
```
Comment (dockerymick) [inline install/perform_scale.adoc]: Would it make sense to move the Network transfer table to the [Hub cluster network configuration](https://github.com/stolostron/rhacm-docs/blob/2.9_stage/networking/network_config_hub.adoc)

[#network-transfer]
=== Network transfer
During tests, network transfer provides stability. See the sizes and network transfer values:
|===
|Size | Inbound network transfer | Outbound network transfer
| 10 clusters
| 6.55 MBs per second
| 5.80 MBs per second
| 20 clusters
| 13.08 MBs per second
| 10.9 MBs per second
|===

File: install/perform_scale.adoc [hunk 9]
```diff
@@ -21,10 +21,12 @@ The maximum number of clusters that {product-title-short} can manage varies base
...
+The following table shows the configuration information for SNO clusters on bare metal environments that were used during this testing:
+//Suggesting to add hardware - Alex 
+//Need to confirm this w/Joydeep | MJ | 09/15/23
+//What info can be provided for 100 clusters or 200 clusters
 |===
-| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region
+| Node | Flavor | vCPU | RAM (GiB) | Disk type | Disk size (GiB) | Count | Region | Tested maximum managed clusters
 
 | Master
 | m5.2xlarge
```
Comment (dockerymick) [inline install/perform_scale.adoc]: Need to confirm the flavor

File: install/perform_scale.adoc [hunk 10]
```diff
@@ -1,10 +1,7 @@
 [#performance-and-scalability]
 = Performance and scalability
 
-{product-title} is tested to determine certain scalability and performance data.
-The major areas that are tested are cluster scalability and search performance.
-
-You can use this information to help you plan your environment.
+{product-title} is tested to determine certain scalability and performance data. The major areas that are tested are cluster scalability and search performance. You can use this information as you plan your environment.
```
Comment (dockerymick) [inline install/perform_scale.adoc:4]: Should the information that Alex presented be added to this file? Let's confirm what information the customer needs.

File: install/cluster_size.adoc [hunk 11]
```diff
@@ -1,6 +1,6 @@
 [#sizing-your-cluster]
 = Sizing your cluster
-
+//are there other platforms that we want to doc here? is this the hardware info that Alex was looking for?| MJ | 09/26/23
```
Comment (dockerymick) [inline install/cluster_size.adoc]:
```suggestion

//are there other platforms that we want to doc here? is this the hardware info that Alex was looking for?| MJ | 09/26/23

//Suggesting to add hardware - Alex : most agreed to add this to Sizing doc: https://github.com/stolostron/rhacm-docs/blob/2b11f255b6a0b2c20819651aa96d95b76253b053/install/cluster_size.adoc 

//libvirt (softwatre on hypervisor) info is important (not vendor), avoid listing vendors share generic numbers

//potentially add chip test metrics to sizing
```

**Comment by dockerymick** [comment]
OCP doc that is similar: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.13/html-single/scalability_and_performance/index#cluster-maximums-environment_object-limits

File: install/perform_scale.adoc [hunk 12]
```diff
@@ -21,35 +21,45 @@ The maximum number of clusters that {product-title-short} can manage varies base
 * Number of resources in the cluster, which depends on factors like the number of policies and applications that are deployed.
 * Configuration of the hub cluster, such as how many pods are used for scaling.
 
-The following table shows the configuration information for the clusters on the Amazon Web Services cloud platform that were used during this testing:
+The managed clusters are {sno} virtual machines hosted on Red Hat Enterprise Linux hypervisors. Virtual machines are used to achieve high-density counts of clusters per single bare metal machine in the testbed. Sushy-emulator is used with libvirt for virtual machines to have an accessible BMC by using Redfish APIs. The following operators are a part of the test installation, Topology Aware Lifecycle Manager, Local Storage Operator, and Red Hat OpenShift GitOps. The following table shows the lab environment scaling information:
 
+.Table for scaling information
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
.Table for environment scaling
```

**Comment by dockerymick** [comment]
@akrzos @bweidenb @bjoydeep Thanks everyone for your help and efforts. I made some updates based on the Google Doc that Alex provided. Will you all please rereview this PR when you have a moment

File: install/perform_scale.adoc [hunk 13]
```diff
@@ -21,35 +21,45 @@ The maximum number of clusters that {product-title-short} can manage varies base
...
-| 32
-| gp2
-| 100
-| 3 or 5 nodes
-| us-east-1
+| 18 GiB
+| 120 GB
 |===
 
 For more information about infrastructure nodes, see, xref:../install/install_connected.adoc#installing-on-infra-node[Installing the {product-title-short} hub cluster on infrastructure nodes]. Also see link:https://docs.openshift.com/container-platform/4.11/machine_management/creating-infrastructure-machinesets.html[Creating infrastructure machine sets].
```
Comment (akrzos) [inline install/perform_scale.adoc]: Line 48/55 regarding infrastructure nodes seems unrelated as no infrastructure nodes were used in this scale testing.

Comment (dockerymick) [reply install/perform_scale.adoc]: Great catch, thanks Alex. i will remove that link reference

Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
```

**Comment by dockerymick** [comment]
@akrzos @bweidenb @bjoydeep when you all have an opportunity, will you re-review this PR

File: install/perform_scale.adoc [hunk 14]
```diff
@@ -1,13 +1,13 @@
 [#performance-and-scalability]
 = Performance and scalability
 
-{product-title} is tested to determine certain scalability and performance data.
-The major areas that are tested are cluster scalability and search performance.
+{product-title} is tested to determine certain scalability and performance data. The major areas that are tested are cluster scalability and search performance. You can use this information as you plan your environment.
 
-You can use this information to help you plan your environment.
+//should we list what we have not tested? - Joydeep | MJ | 09/28/23
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
```

File: install/perform_scale.adoc [hunk 15]
```diff
@@ -21,35 +20,35 @@ The maximum number of clusters that {product-title-short} can manage varies base
 * Number of resources in the cluster, which depends on factors like the number of policies and applications that are deployed.
 * Configuration of the hub cluster, such as how many pods are used for scaling.
 
-The following table shows the configuration information for the clusters on the Amazon Web Services cloud platform that were used during this testing:
+The managed clusters are {sno} virtual machines hosted on Red Hat Enterprise Linux hypervisors. Virtual machines are used to achieve high-density counts of clusters per single bare metal machine in the testbed. Sushy-emulator is used with libvirt for virtual machines to have an accessible BMC by using Redfish APIs. The following operators are a part of the test installation, Topology Aware Lifecycle Manager, Local Storage Operator, and Red Hat OpenShift GitOps. The following table shows the lab environment scaling information:
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
The managed clusters are {sno} virtual machines hosted on Red Hat Enterprise Linux hypervisors. Virtual machines are used to achieve high-density counts of clusters per single bare metal machine in the testbed. Sushy-emulator is used with libvirt for virtual machines to have an accessible bare metal cluster by using Redfish APIs. The following operators are a part of the test installation, Topology Aware Lifecycle Manager, Local Storage Operator, and Red Hat OpenShift GitOps. The following table shows the lab environment scaling information:
```

File: install/perform_scale.adoc [hunk 16]
```diff
@@ -21,35 +20,35 @@ The maximum number of clusters that {product-title-short} can manage varies base
...
+| 18 GiB
+| 120 GB
 |===
 
-For more information about infrastructure nodes, see, xref:../install/install_connected.adoc#installing-on-infra-node[Installing the {product-title-short} hub cluster on infrastructure nodes]. Also see link:https://docs.openshift.com/container-platform/4.11/machine_management/creating-infrastructure-machinesets.html[Creating infrastructure machine sets].
 
 [#search-scalability]
 == Search scalability
 
+//team reviewing info and tracking this on issue https://issues.redhat.com/browse/ACM-7640; //What recommendation do we want to give to the customers? and at what interval numbers?  | MJ | 09/26
```
Comment (dockerymick) [inline install/perform_scale.adoc]:
```suggestion
```

**Comment by dockerymick** [comment]
Approval from PM @bweidenb 
Screenshot 2023-10-18 at 10 19 01 AM

**Review by swopebe — APPROVED** [review]

**Comment by dockerymick** [comment]
Thanks everyone for the review and approvals @swopebe @akrzos @bweidenb
